### PR TITLE
Add comments to intermediates

### PR DESCRIPTION
**Pull request by tilleryd**
[sc-20140]

### CODE REVIEW

File: gm/intermediates.cue [hunk 1]
```diff
@@ -13,19 +13,37 @@ import (
...
 // "Functions" for Grey Matter config objects with defaults
-// representing an ingress to a local service. Override-able.
+// reflecting common local service configuration. Override-able.
 /////////////////////////////////////////////////////////////
 
+// This represents a greymatter domain object, which deals with incoming 
+// requests to the service.
+// See https://docs.greymatter.io/service-mesh/domain/ for more details.
 #domain: greymatter.#Domain & {
+	// Enable this to turn on secure traffic to the service.
```
Comment (dontlaugh) [inline gm/intermediates.cue:23]:
```suggestion
	// Set _force_https to true to turn on secure traffic to the service.
```

File: gm/intermediates.cue [hunk 2]
```diff
@@ -13,19 +13,37 @@ import (
...
+	// Configures TLS settings for incoming requests, utilizing 
+	// mounted certificates to allow for HTTPS traffic
+	force_https: _force_https
 	if _force_https == true {
 		ssl_config: greymatter.#SSLConfig & {
+			// Specify a TLS Protocol to use when communicating
+			// Supported options are:
+			// TLS_AUTO TLSv1_0 TLSv1_1 TLSv1_2 TLSv1_3
 			protocols: [ "TLSv1_2"]
+			// Details for the certificate used to allow secure traffic to flow
```
Comment (dontlaugh) [inline gm/intermediates.cue:42]:
```suggestion
			// TLS certificate defaults, if enabled.
```

File: gm/intermediates.cue [hunk 3]
```diff
@@ -37,72 +55,131 @@ import (
...
-	_tcp_upstream?:              string        // for TCP listeners, you can just specify the upstream cluster
-	_is_ingress:                 bool | *false // specifiy if this listener is for ingress which will active default HTTP filters
-	_gm_observables_topic:       string        // unique topic name for observable audit collection
-	_spire_self:                 string        // can specify current identity - defaults to "edge"
-	_spire_other:                string        // can specify an allowable downstream identity - defaults to "edge"
+	// For TCP listeners, you can just specify the upstream cluster
+	_tcp_upstream?: string
+	// Specifiy if this listener is for ingress, which will activate default HTTP filters
+	_is_ingress: bool | *false
+	// Unique topic name for observable audit collection
```
Comment (dontlaugh) [inline gm/intermediates.cue:67]: Add a link to an explanation of "observables" here.

cc @annac-gm the word "observables" is a term we've invented and I don't think anyone outside of this company knows what it is.

File: gm/intermediates.cue [hunk 4]
```diff
@@ -125,10 +204,19 @@ import (
 			if _enable_rbac {
 				"envoy.rbac"
 			},
+			// This filter is essential to the operation of the mesh and should
+			// not be disabled.
 			"gm.metrics",
 			...string,
 		]
+
+		// This object contains the configuration for filters potentially applied
```
Comment (dontlaugh) [inline gm/intermediates.cue:213]:
```suggestion
		// htp_filters contains the configuration for filters potentially applied
```

File: gm/intermediates.cue [hunk 5]
```diff
@@ -37,72 +55,131 @@ import (
...
 				stat_prefix: _tcp_upstream
 			}
 		}
 	}
 
-	// if there isn't a tcp cluster, then assume http filters, and provide the usual defaults
+	// Non-TCP listeners that are ingress can have HTTP filters applied to them
 	if _tcp_upstream == _|_ && _is_ingress == true {
+
+		// This list contains the filters which will be active on the listener.
```
Comment (dontlaugh) [inline gm/intermediates.cue:163]:
```suggestion
		// The active_http_filters list contains the filters which will be active on the listener.
```

**Review by dontlaugh — CHANGES_REQUESTED** [review]
A few changes. Looks good overall.

Avoid "this" and "that" when introducing a big complicated stanza. Call the stanza by name.

File: gm/intermediates.cue [hunk 6]
```diff
@@ -282,7 +381,7 @@ import (
 		circuit_breakers: #circuit_breaker // can specify circuit breaker levels for normal
 		// and high priority traffic with configured defaults
 	}
-	if _load_balancer != _|_ {
+	if _load_balancer != _|_ {// Allows for configuration of a load balancer, designated by the policy type
```
Comment (dontlaugh) [inline gm/intermediates.cue:384]: Don't introduce more trailing comments. They're harder to maintain.
```suggestion
        // Allows for configuration of a load balancer, designated by the policy type
	if _load_balancer != _|_ {
```

File: gm/intermediates.cue [hunk 7]
```diff
@@ -13,19 +13,37 @@ import (
...
+	zone_key: mesh.spec.zone
+	// Configures TLS settings for incoming requests, utilizing 
+	// mounted certificates to allow for HTTPS traffic
+	force_https: _force_https
 	if _force_https == true {
 		ssl_config: greymatter.#SSLConfig & {
+			// Specify a TLS Protocol to use when communicating
+			// Supported options are:
+			// TLS_AUTO TLSv1_0 TLSv1_1 TLSv1_2 TLSv1_3
 			protocols: [ "TLSv1_2"]
```
Comment (alecholmez) [inline gm/intermediates.cue:41]: Should we not default to auto? I think that's what envoy does

File: gm/intermediates.cue [hunk 8]
```diff
@@ -37,72 +55,131 @@ import (
...
+		// This list contains the filters which will be active on the listener.
+		// NB: Even if configuration exists in http_filters for a filter,
+		// only filters which are listed as active will be applied and used.
+		// To add custom filters, introduce a new toggle above and match the
+		// pattern below to place it correctly in the chain. 
 		active_http_filters: [
 			if _enable_fault_injection {
 				"envoy.fault"
 			},
+			// Note: Inheaders and impersonation only function when mTLS is active
```
Comment (alecholmez) [inline gm/intermediates.cue:172]: Probably want to note that inheaders and impersonation will never be set on the same proxy

File: gm/intermediates.cue [hunk 9]
```diff
@@ -305,7 +404,7 @@ import (
 	high?: #circuit_breaker_default
 }
 
-#circuit_breaker_default: {
+#circuit_breaker_default: {// Default values taken from the Envoy documentation
```
Comment (alecholmez) [inline gm/intermediates.cue:407]:
```suggestion
#circuit_breaker_default: { // Default values taken from the Envoy documentation
```

File: gm/intermediates.cue [hunk 10]
```diff
@@ -367,7 +466,7 @@ import (
 
 // Allows for RBAC permissions to be applied to a service and its configuration
 #envoy_rbac_filter: rbac.#RBAC | *#default_rbac
-#default_rbac: {
+#default_rbac: {// This allows all traffic to flow with no restriction
```
Comment (alecholmez) [inline gm/intermediates.cue:469]:
```suggestion
#default_rbac: { // This allows all traffic to flow with no restriction
```

File: gm/intermediates.cue [hunk 11]
```diff
@@ -420,7 +519,7 @@ import (
 // used in other contexts, such as RBAC configurations.
 #envoy_jwt_authn: jwt_authn.#JwtAuthentication & {
 	providers: {
-		keycloak?: {
+		keycloak?: {//If Using another provider name below, change this field in your service
```
Comment (alecholmez) [inline gm/intermediates.cue:522]:
```suggestion
		keycloak?: { //If Using another provider name below, change this field in your service
```

File: gm/intermediates.cue (hunk 7 above)
Comment (tilleryd) [reply gm/intermediates.cue]: I made the change and tested in a deployment and TLS at the edge worked fine with "TLS_AUTO".

**Review by alecholmez — APPROVED** [review]

**Review by dontlaugh — APPROVED** [review]

**Pull request merged**